components/icon-settings/index.jsx

```diff
@@ -6,7 +6,14 @@ import PropTypes from 'prop-types';
 
 class IconSettings extends React.Component {
 	getChildContext () {
-		return { iconPath: this.props.iconPath };
+		return {
+			iconPath: this.props.iconPath,
+			actionSprite: this.props.actionSprite,
```

Comment:
I like this. Kind of aligns with the [anti-concatenation approach](https://github.com/salesforce/design-system-react/blob/master/docs/codebase-overview.md#all-text-can-be-internationalized) of this library.

Reply:
* Do you think there's a way to make it more obvious that it's the path + file name. The `propType` is a string, so that should help. 
* Also could you add an example/storybook story that uses this in the `examples/icon` folder?

* I know we failed to have already didn't have a propTypes object on `IconSettings`. Could you add the props you added to a `IconSettings.propTypes` object add comment descriptions for them. Please look at other components for examples.

I think this will be helpful for devs, so they can put their icons anyway they want.

Reply:
I'll pull this branch down on Monday and make sure it works with the story I've requested. Thank you for your contribution. I'll try to get a release out for you next week. It will mostly likely be 0.8.0 that uses the new positioning library, PopperJS.

Reply:
Thanks @interactivellama for the comments

I've added:
- an Icon Settings example.
- the new icons setting option to the readme.
- the prop types to icon settings.

Note that I've added a new constant `SLDSIconSettings` so this is what is rendered as the element tag.

Jest Snapshots have been updated.

With regards to the sprite properties on the icon settings, it could be named `actionSpritePath` but that may be even more confusing cause in reality you are passing an svg object and webpack is extracting that out and replacing it with the path string. I think with an icon settings example and the documentation it should be clear enough to devs what to do.

Let me know if there are any other changes that need to be done.